core/src/main/java/org/apache/iceberg/CachingCatalog.java

```diff
@@ -164,19 +164,20 @@ public Table loadTable(TableIdentifier ident) {
   @Override
   public boolean dropTable(TableIdentifier ident, boolean purge) {
     boolean dropped = catalog.dropTable(ident, purge);
-    invalidate(canonicalizeIdentifier(ident));
+    invalidate(ident);
     return dropped;
   }
 
   @Override
   public void renameTable(TableIdentifier from, TableIdentifier to) {
     catalog.renameTable(from, to);
-    invalidate(canonicalizeIdentifier(from));
+    invalidate(from);
   }
 
-  private void invalidate(TableIdentifier ident) {
-    tableCache.invalidate(ident);
-    tableCache.invalidateAll(metadataTableIdentifiers(ident));
+  public void invalidate(TableIdentifier ident) {
```

Comment:
Why is this now public?

Reply:
Used by `SparkCatalog` to forcibly invalidate cache entry.